spyder/plugins/maininterpreter.py

```diff
@@ -152,6 +161,7 @@ def python_executable_changed(self, pyexec):
                       "make sure to select a valid one."), QMessageBox.Ok)
             self.pyexec_edit.setText(def_pyexec)
             return
+        return True

```

Comment:
If you make this method to `return True` at the end, then you need to change the behavior of the previous `return`'s to be `return False`, and not simply `return` (as it right now). That's to be consistent.